gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/ControllerUserDefinedMessageHandlerFactory.java

```diff
@@ -51,7 +51,7 @@
 class ControllerUserDefinedMessageHandlerFactory implements MessageHandlerFactory {
   private boolean flowCatalogLocalCommit;
   private GobblinServiceJobScheduler jobScheduler;
-  private GobblinServiceFlowConfigResourceHandler resourceHandler;
+  private FlowConfigsResourceHandler resourceHandler;
```

Comment:
Instead of injecting a specific implementation, we inject an interface. GobblinServiceFlowConfigResourceHandler  implements this FlowConfigsResourceHandler.

It's generally a good practice to use interface-based dependency injection everywhere. That will allow us to replace implementations completely, and its more convenient to mock in unit tests. 

For example, some of our classes inherit from Guava service class. And that service class has final methods. Mockito by default cannot override final methods, so when we try to mock such dependency, part of the calls go to the mocked methods, while others go to real implementation, causing complicated bugs. If the dependency is an interface, we can mock it fully.